### PR TITLE
Handle Sentry HTTP errors

### LINKED ISSUE #21: Exporter queries Sentry API too fast and crashes
**Describe the bug**
When querying the `/metrics` endpoints with 11 sentry projects, I get a 500 error with the exception `AttributeError: 'str' object has no attribute 'get'` raised in `helpers/prometheus.py`, line 257 (`events_1h += int(issue.get("count") or 0)`).

When printing `project_issues_1h`, it contains the following:
```
{'detail': 'You are attempting to use this endpoint too quickly. Limit is 3/1s'}
```

**To Reproduce**
1. Add at least 11 projects on Sentry
2. Query the metrics endpoint

**Expected behavior**
No crash

**Screenshots**
image

image

**Desktop (please complete the following information):**
 - Does not apply

**Smartphone (please complete the following information):**
 - Does not apply

**Additional context**
Does not apply

### PR DESCRIPTION
**Pull request by lululombard**
## Description

This adds a retry module to retry requests to close #21 

PS: I removed `from os import get_env` is because `get_env` wasn't used anywhere

### Before

image
image

### After

image
image

## Checklist

- [X] All tests are passing
- [X] New tests were created to address changes in pr (and tests are passing)
- [X] Updated README and/or documentation, if necessary

Thanks for contributing!

### CODE REVIEW

**Comment by lululombard** [comment]
Hey @italux, any plans to review this? I'd like to use this fix and our Kubernetes setup relies on your Docker image, thanks!

File: libs/sentry.py [hunk 1]
```diff
@@ -27,9 +27,12 @@ def __init__(self, base_url, auth_token):
         self.__token = auth_token
         self.__session = requests.Session()
 
+    @retry(requests.exceptions.HTTPError, tries=2, delay=1)
```
Comment (italux) [inline libs/sentry.py:30]: @lululombard Could you make these values (`tries` & `delay`) configurable as proposed on #23 ?

Comment (lululombard) [reply libs/sentry.py:30]: Will do!

**Comment by italux** [comment]
> Hey @italux, any plans to review this? I'd like to use this fix and our Kubernetes setup relies on your Docker image, thanks!

Hey @lululombard , first of all, thanks for identity and contributing to fixing it. I've added a small comment which I think worth to address and make the configuration explicit, also would be good to update the README file Limitations section with the rate limit information as @gonzalosr did on #23

File: libs/sentry.py (hunk 1 above)
Comment (italux) [reply libs/sentry.py:30]: Another thing that would be good is to retry only on `HTTP 429` instead of retry on any HTTP error `5xx` as mentioned on https://github.com/italux/sentry-prometheus-exporter/pull/23#issuecomment-922543930, retrying on any HTTP error can make debugging harder, what you think?

Maybe you should also add a backoff here, something like:
```suggestion
    @retry(requests.exceptions.HTTPError, tries=2, delay=1, backoff=2, jitter=0.5)
```

Comment (lululombard) [reply libs/sentry.py:30]: The backoff and jitter is a good idea, for the handling of exceptions, I think it's fine to retry on anything as exceptions are logged with the error code [cf screenshot](https://user-images.githubusercontent.com/2182934/132733103-b0232917-a5ee-443d-adc4-7755f474534c.png)

**Comment by lululombard** [comment]
@italux Thanks for the review, I'll spend some time today to address what you raised :)

**Comment by lululombard** [comment]
@italux everything should be fine now, I also upgraded Flask and tested it with success :)

File: README.md [hunk 2]
```diff
@@ -84,6 +84,20 @@ export SENTRY_ISSUES_14D=False
...
+You can tweak your settings with environment variables, though default settings should work:
+
+|  Environment variable    | Value type | Default value |                         Purpose                         |
+|:------------------------:|:----------:|:-------------:|:-------------------------------------------------------:|
+| `SENTRY_RETRY_TRIES`     | Integer    | 3             | How many retries should be made in case of an exception |
+| `SENTRY_RETRY_DELAY`     | Float      | 1             | How many seconds to wait between retries                |
+| `SENTRY_RETRY_MAX_DELAY` | Float      | 10            | Max delay to wait between retries                       |
+| `SENTRY_RETRY_BACKOFF`   | Float      | 2             | Multiplier applied to delay between attempts            |
+| `SENTRY_RETRY_JITTER`    | Float      | 0.5           | Extra seconds added to delay between attempts           |
+
```
Comment (italux) [inline README.md:100]: Could you please move this section to start at line `107` since this is a workaround to handle a known limitation on Sentry API?

File: README.md [hunk 3]
```diff
@@ -84,6 +84,20 @@ export SENTRY_ISSUES_14D=False
 ```
 As with `SENTRY_AUTH_TOKEN`, all of these variables can be passed in through the `docker run -e VAR_NAME=<>` command or via the `.env` file if using Docker Compose.
 
+### Retries to get around limits
```
Comment (italux) [inline README.md:87]: Small suggestion on the title
```suggestion
### Sentry API retry calls
```

File: requirements.txt [hunk 4]
```diff
@@ -1,3 +1,4 @@
-Flask==1.0.2
+Flask==2.0.1
```
Comment (italux) [inline requirements.txt:1]: Thanks for validating the new version! But I'd like to keep this change into a different PR and this upgrade will be addressed on #26 
```suggestion
Flask==1.0.2
```

**Comment by italux** [comment]
> @italux everything should be fine now, I also upgraded Flask and tested it with success :)

@lululombard Thanks for all changes, I've added some small comments, which aren't a blocker. So I'll approve the PR anyway and generate a release tomorrow (BRT) so if you're able to fix would be nice.

**Review by italux — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (lululombard) [reply README.md]: Thanks, applied

File: requirements.txt (hunk 4 above)
Comment (lululombard) [reply requirements.txt]: Sounds good, I reverted it :)

File: README.md (hunk 2 above)
Comment (lululombard) [reply README.md]: I wasn't sure if you meant line 107 as it was before or with the block, but it's been moved to the `## ℹ️ Important Notes` section, after `### Limitations`, if that's not right let me know and I'll change it :)

Comment (italux) [reply README.md]: It's fine where you put, thanks!

**Comment by italux** [comment]
@lululombard New [release generated](https://github.com/italux/sentry-prometheus-exporter/releases/tag/v0.3.0) and [docker image](https://hub.docker.com/repository/docker/italux/sentry-prometheus-exporter/tags?page=1&ordering=name) updated

**Comment by lululombard** [comment]
Thank you @italux, I deployed the image on our Kubernetes cluster and everything is now working fine :) 
image
